File: Day40/RotateString.py
class Solution(object):
    def rotateString(self, s, goal):
        if len(s) != len(goal):
            return False
        doubled = s + s
        if goal in doubled:
            return True
        else:
            return False

    @staticmethod
    def main():
        s = input().strip()
        goal = input().strip()
        sol = Solution()
        print(sol.rotateString(s, goal))

if __name__ == "__main__":
    Solution.main()

File: Day40/RotateString.java
import java.util.*;

public class RotateString {
    public boolean rotateString(String s, String goal) {
        if (s.length() != goal.length()) return false;
        String doubled = s + s;
        return doubled.contains(goal);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String s = sc.nextLine().trim();
        String goal = sc.nextLine().trim();
        RotateString sol = new RotateString();
        System.out.println(sol.rotateString(s, goal));
    }
}
